src/gt4py/next/iterator/transforms/infer_domain.py

```diff
@@ -362,30 +405,35 @@ def infer_expr(
     - domain: The domain `expr` is read at.
     - symbolic_domain_sizes: A dictionary mapping axes names, e.g., `I`, `Vertex`, to a symbol
       name that evaluates to the length of that axis.
+    - allow_uninferred: Allow `as_fieldop` expressions whose domain is either unknown (e.g.
+      because of a dynamic shift) or empty.
```

Comment:
"empty" is used only here, probably use a formulation with "never accessed" here